core/config/toml/types.go

```diff
@@ -1494,6 +1502,31 @@ func (t *Tracing) ValidateConfig() (err error) {
 		}
 	}
 
+	if t.TLSCertPath != nil {
+		ok := isValidFilePath(*t.TLSCertPath)
+		if !ok {
+			err = multierr.Append(err, configutils.ErrInvalid{Name: "TLSCertPath", Value: *t.TLSCertPath, Msg: "must be a valid file path"})
+		}
+	}
+
+	if t.Mode != nil {
+		switch *t.Mode {
+		case "secure":
+			// TLSCertPath must be set
+			if t.TLSCertPath == nil {
+				err = multierr.Append(err, configutils.ErrMissing{Name: "TLSCertPath", Msg: "must be set when Tracing.Mode is secure"})
+			}
+		case "insecure":
```

Comment:
Should insecure only be allowed for dev/test mode? and disallowed in production builds?

Reply:
There are valid use cases for allowing insecure mode in production builds, so I'd be cautious to assert that condition